File: pset6/readability/readability.py
from cs50 import get_string


letters = 0 # total number of words in string
sentences = 0 # total number of sentences in string
words = 1

frase = get_string("Text: ")

for i in range(0, len(frase)):
    if frase[i] == '.' or frase[i] == '!' or frase[i] == '?':
        sentences += 1
    if frase[i].isalnum():
        letters += 1
    if frase[i] == ' ':
        words += 1

L = int((letters * 100.0) / words) # average number of letters per 100 words
S = int((sentences * 100.0) / words)  # average number of sentences per 100 words
X = round(0.0588 * L - 0.296 * S - 15.8)


if X < 1:
    print("Before Grade 1")
elif X > 16:
    print("Grade 16")
else:
    print(f"Grade {X}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <ctype.h>
#include <string.h>

int main(void)
{

    string texto = get_string("Text: ");

    float sentences = 0;
    float words = 1;
    float letters = 0;
    string vazio = " ";

    for (int i = 0, n = strlen(texto); i < n; i++)
    {
        if (isspace(texto[i]))
        {
            words++;
        }

        if (texto[i] == '.' || texto[i] == '!' || texto[i] == '?')
        {
            sentences++;
        }

        if (isalpha(texto[i]))
        {
            letters++;
        }

    }

    float L = letters / words * 100;
    float S = sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int resultado = round(index);

    if (resultado >= 1 && resultado <= 16)
    {
        printf("Grade %i\n", resultado);
    }
    else
    {
        if (resultado < 1)
        {
            printf("Before Grade 1\n");
        }
        if (resultado > 16)
        {
            printf("Grade 16+\n");
        }
    }
}